server/core/src/voting.cpp

```diff
@@ -247,4 +248,10 @@ float calculate(
     return vote;
 } // calculate
 
+auto vote_is_zero(float _vote) -> bool
+{
+    constexpr double epsilon = 0.00000001;
+    return _vote - epsilon <= vote::zero;
+} // vote_is_zero
```

Comment:
So this is how we've always checked for zero, it's just that you've wrapped the logic so that it is in one location, correct?

Reply:
Yes and no.

This is the way we checked for zero in several places in `irods_resource_redirect.cpp`. In other places, the vote value was compared directly against `0.0f` (or, more recently, `vote::zero`). You can see the specific instances elsewhere in the diff.

This function provides a thin abstraction so that the implementation can be swapped out later if we so choose. I think comparing against zero should be consistent, in any case.

Reply:
All of these comparisons have already been replaced by calls to `vote_is_zero` in this PR. I can put them back and make a new issue to address this at a later time if that seems better. Or are you referring to all floating point comparisons in the codebase?

Reply:
No need to revert anything.

I was under the impression that there were other places outside of your PR that dealt with voting, but compared floats incorrectly. It sounds like you've already addressed all known locations though.

Reply:
Oh, I understand now. Yeah, I didn't really look outside of `irods_resource_redirect.cpp`. Looks like some of the resource plugins are making direct comparisons, so I made an issue to address those places: https://github.com/irods/irods/issues/6932

Reply:
Cool cool